### PR TITLE
Policy Added for control id Azure_APIManagement_AuthN_Use_AAD

### PR DESCRIPTION
**Pull request by jaiswalricha**
Policy Added for  Control id "Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN"

### CODE REVIEW

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+[Security - APIManagement - Deny APIs with other than AAD Identity Provider Enabled](Security%20-%20APIManagement%20-%20Deny%20APIs%20with%20other%20than%20AAD%20Identity%20Provider%20Enabled.json)
+
+#### Parameter details
+Param Name|Description|Default Value|Mandatory?
+|----|----|----|----|
+| Effect | The effect determines what happens when the policy rule is evaluated to match| Audit |No |
+
+### Notes
+1. Policy will flag/target individual APIs(Microsoft.ApiManagement/service/apis) in API Management service.
```
Comment (TarunKrShukla) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md:40]: Is this note applicable for both the above policies, please mention that.
NIT: APIs (Microsoft.ApiManagement/service/apis)

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+## Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN 
+
+### DisplayName
```
Comment (TarunKrShukla) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md:3]: There is one index readme.md file where we update all list of controls for which policy is supported we  need to add this there as well. https://github.com/azsk/AzTS-docs/tree/users/v-rijaiswal/CustomPolicyValidation/Policies

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny APIs with Basic Authentication Enabled.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      "allOf": [
+        {
+          "field": "type",
+          "equals": "Microsoft.ApiManagement/service/portalsettings"
+        },
+        {
+          "allOf": [
+            {
+              "field": "name",
+              "contains": "signup"
```
Comment (TarunKrShukla) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny APIs with Basic Authentication Enabled.json:14]: should we check for equals?

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny APIs with Basic Authentication Enabled.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{
+  "mode": "All",
+  "policyRule": {
+    "if": {
+      "allOf": [
+        {
```
Comment (TarunKrShukla) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny APIs with Basic Authentication Enabled.json:6]: Policy will generate policy results against which resource type?

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny APIs with other than AAD Identity Provider Enabled.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
+{
+  "mode": "All",
+  "policyRule": {
+    "if": {
+      "allOf": [
```
Comment (TarunKrShukla) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny APIs with other than AAD Identity Provider Enabled.json:5]: Policy will generate policy results against which resource type?

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md (hunk 1 above)
Comment (jaiswalricha) [reply Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md:40]: corrected the statement for both policies.

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny API Management Services with Basic Authentication Enabled.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    "if": {
+      "allOf": [
+        {
+          "field": "type",
+          "equals": "Microsoft.ApiManagement/service/portalsettings"
+        },
+        {
+          "allOf": [
+            {
+              "field": "name",
```
Comment (saurabhgautam-msft) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny API Management Services with Basic Authentication Enabled.json:13]: why two allof ? All the conditions could reside under single allof right?

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny API Management Services with other than AAD Identity Provider Enabled.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  "mode": "All",
+  "policyRule": {
+    "if": {
+      "allOf": [
+        {
+          "field": "type",
+          "equals": "Microsoft.ApiManagement/service/identityProviders"
+        },
+        {
+          "anyOf": [
```
Comment (saurabhgautam-msft) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny API Management Services with other than AAD Identity Provider Enabled.json:11]: there is no other condition so, no need for anyOf right?

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny API Management Services with Basic Authentication Enabled.json (hunk 6 above)
Comment (saurabhgautam-msft) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny API Management Services with Basic Authentication Enabled.json:13]: is 'name' an alias?

Comment (jaiswalricha) [reply Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Security - APIManagement - Deny API Management Services with Basic Authentication Enabled.json:13]: Yes Saurabh

**Review by saurabhgautam-msft — APPROVED** [review]
LGTM

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md (hunk 2 above)
Comment (TarunKrShukla) [reply Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md:3]: this comment not fixed yet?

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+Control can be covered with the below mentioned Azure policy:
+- Policy to validate if Basic Authentication is enabled for API Management.
+- Policy to validate if Identity Provider is enabled other than Azure Active Directory.
+
+#### Policy Details
+
+Following policy can be used to validate if Basic Authentication is enabled.
+
+#### Policy Definition
+[Security - APIManagement - Deny API Management Services with Basic Authentication Enabled](Security%20-%20APIManagement%20-%20Deny%20APIs%20with%20Basic%20Authentication%20Enabled.json)
```
Comment (TarunKrShukla) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md:16]: plz check the links

File: Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+___ 
+
+
+#### Policy Details
+
+Following policy can be used to validate if Identity Provider is enabled other than Azure Active Directory.
+
+#### Policy Definition
+
+[Security - APIManagement - Deny API Management Services with other than AAD Identity Provider Enabled](Security%20-%20APIManagement%20-%20Deny%20APIs%20with%20other%20than%20AAD%20Identity%20Provider%20Enabled.json)
```
Comment (TarunKrShukla) [inline Policies/APIManagement/Azure_APIManagement_AuthN_Use_AAD_for_Client_AuthN/Readme.md:32]: plz check the links

**Review by TarunKrShukla — APPROVED** [review]
Looks good.

**Pull request merged**